src/callbacks/steady_state.jl

```diff
@@ -0,0 +1,175 @@
+                                        <:PeriodicCallbackAffect{<:SteadyStateCallback}})
+    @nospecialize cb # reduce precompilation time
+
+    cb_ = cb.affect!.affect!
+
+    print(io, "SteadyStateCallback(abstol=", cb_.abstol, ", reltol=", cb_.reltol, ")")
+end
+
+function Base.show(io::IO, ::MIME"text/plain",
+                   cb::DiscreteCallback{<:Any, <:SteadyStateCallback})
+    @nospecialize cb # reduce precompilation time
+
+    if get(io, :compact, false)
+        show(io, cb)
+    else
+        cb_ = cb.affect!
+
+        setup = ["absolute tolerance" => cb_.abstol,
+            "relative tolerance" => cb_.reltol,
+            "interval" => cb_.interval,
+            "interval size" => cb_.interval_size]
+        summary_box(io, "SteadyStateCallback", setup)
+    end
+end
+
+function Base.show(io::IO, ::MIME"text/plain",
+                   cb::DiscreteCallback{<:Any,
+                                        <:PeriodicCallbackAffect{<:SteadyStateCallback}})
+    @nospecialize cb # reduce precompilation time
+
+    if get(io, :compact, false)
+        show(io, cb)
+    else
+        cb_ = cb.affect!.affect!
+
+        setup = ["absolute tolerance" => cb_.abstol,
+            "relative tolerance" => cb_.reltol,
+            "interval" => cb_.interval,
+            "interval_size" => cb_.interval_size]
+        summary_box(io, "SteadyStateCallback", setup)
+    end
+end
+
+# `affect!` (`PeriodicCallback`)
+function (cb::SteadyStateCallback)(integrator)
+    steady_state_condition(cb, integrator) || return nothing
+
+    print_summary(integrator)
+
+    terminate!(integrator)
+end
+
+# `affect!` (`DiscreteCallback`)
+function (cb::SteadyStateCallback{Int})(integrator)
+    print_summary(integrator)
+
+    terminate!(integrator)
+end
+
+# `condition` (`DiscreteCallback`)
+function (steady_state_callback::SteadyStateCallback)(vu_ode, t, integrator)
+    return steady_state_condition(steady_state_callback, integrator)
+end
+
+@inline function steady_state_condition(cb, integrator)
+    (; abstol, reltol, previous_ekin, interval_size) = cb
+
+    vu_ode = integrator.u
+    v_ode, u_ode = vu_ode.x
+    semi = integrator.p
+
+    # Calculate kinetic energy
+    ekin = 0.0
+    foreach_system(semi) do system
+        v = wrap_v(v_ode, system, semi)
+
+        # Calculate kintetic energy
+        for particle in each_moving_particle(system)
+            velocity = current_velocity(v, system, particle)
+            ekin += 0.5 * system.mass[particle] * dot(velocity, velocity)
+        end
+    end
+
+    push!(previous_ekin, ekin)
+
+    terminate = false
+
+    if interval_size_condition(cb, integrator)
+        popfirst!(previous_ekin)
+
+        # Calculate MSE only over the `interval_size`
+        mse = 0.0
+        for index in 1:interval_size
+            mse += (previous_ekin[index] - ekin)^2
+        end
+
+        mse /= interval_size
+
+        threshold = abstol + reltol * ekin
+
+        terminate = mse <= threshold
+    end
+
```

Comment:
```suggestion
# Add current kinetic energy       
push!(previous_ekin, ekin)
```

Reply:
Why do you want to do this after computing the MSE?
Why not
```
if too large
    popfirst!(previous_ekin)
end

push!(previous_ekin, ekin)
```

Reply:
Because we don't have to compare the current value with the current value because that will be 0.